packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -142,11 +142,27 @@ class Directionality extends InheritedWidget {
 /// This is more efficient than adding and removing the child widget from the
 /// tree on demand.
 ///
+/// ## Opacity Animation
+///
+/// [Opacity] animations should be built using [AnimatedOpacity] rather than
+/// manually rebuilding the [Opacity] widget.
+///
+/// Animating an [Opacity] widget directly causes the widget (and possibly its
+/// subtree) to rebuild each frame, which is not very efficient. Consider using
+/// an [AnimatedOpacity] instead. [AnimatedOpacity] uses a
```

Comment:
We've ended up with two paragraphs that say the same thing. :-)

Reply:
I'll fix it when I get back to this ticket.  Currently this ticket depends on #15466 which depends on #18173